Gems/ONNX/Code/Source/Clients/Model.cpp

```diff
@@ -0,0 +1,153 @@
+        }
+        else
+        {
+            m_modelName = initSettings.m_modelName;
+        }
+
+        m_modelColor = initSettings.m_modelColor;
+
+        // Grabs environment created on init of system component.
+        Ort::Env* env = nullptr;
+        ONNXRequestBus::BroadcastResult(env, &ONNXRequestBus::Events::GetEnv);
+
+#ifdef ENABLE_CUDA
+        // OrtCudaProviderOptions must be added to the session options to specify execution on CUDA.
+        // Can specify a number of parameters about the CUDA execution here - currently all left at default.
+        Ort::SessionOptions sessionOptions;
+        if (initSettings.m_cudaEnable)
+        {
+            OrtCUDAProviderOptions cuda_options;
+            sessionOptions.AppendExecutionProvider_CUDA(cuda_options);
+        }
+        m_cudaEnable = initSettings.m_cudaEnable;
+#endif
+
+        // The model_path provided to Ort::Session needs to be const wchar_t*, even though the docs state const char* - doesn't work otherwise.
+        AZStd::string onnxModelPathString = onnxModelPath.String();
+        m_session = Ort::Session(*env, AZStd::wstring(onnxModelPathString.cbegin(), onnxModelPathString.cend()).c_str(), sessionOptions);
+        m_memoryInfo = Ort::MemoryInfo::CreateCpu(OrtDeviceAllocator, OrtMemTypeCPU);
+
+        // Grabs memory allocator created on init of system component.
+        Ort::AllocatorWithDefaultOptions* m_allocator;
+        ONNXRequestBus::BroadcastResult(m_allocator, &ONNXRequestBus::Events::GetAllocator);
+
+        // Extract input names from model file and put into const char* vectors.
+        // Extract input shapes from model file and put into AZStd::vector<int64_t>.
+        m_inputCount = m_session.GetInputCount();
+        for (size_t i = 0; i < m_inputCount; i++)
+        {
+            const char* in_name = m_session.GetInputName(i, *m_allocator);
+            m_inputNames.push_back(in_name);
+
+            std::vector<int64_t> inputShape = m_session.GetInputTypeInfo(i).GetTensorTypeAndShapeInfo().GetShape();
+            AZStd::vector<int64_t> azInputShape(inputShape.begin(), inputShape.end());
+            for (int index = 0; index < azInputShape.size(); index++)
+            {
+                if (azInputShape[index] == -1)
+                {
+                    azInputShape[index] = 1;
+                }
+            }
+            m_inputShapes.push_back(azInputShape);
+        }
+
+        // Extract output names from model file and put into const char* vectors.
+        // Extract output shapes from model file and put into AZStd::vector<int64_t>.
+        // Initialize m_outputs vector using output shape and count.
+        m_outputCount = m_session.GetOutputCount();
+        AZStd::vector<AZStd::vector<float>> outputs(m_outputCount);
+        for (size_t i = 0; i < m_outputCount; i++)
+        {
+            const char* out_name = m_session.GetOutputName(i, *m_allocator);
```

Comment:
standards: [lowerCamelCase](https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#naming) is the standard.

Reply:
Changed